lib/octokit/client/stats.rb

```diff
@@ -1,106 +1,54 @@
-      #   @client.code_frequency_stats('octokit/octokit.rb')
-      def code_frequency_stats(repo, options = {})
-        get_stats(repo, "code_frequency", options)
-      end
-
-      # Get the weekly commit count for the repo owner and everyone else
+      # Get contributors list with additions, deletions, and commit counts
       #
-      # @param repo [Integer, String, Hash, Repository] A GitHub repository
-      # @option retry_timeout [Number] How long Octokit should keep trying to get stats (in seconds)
-      # @option retry_wait [Number] How long Octokit should wait between retries.
-      # @return [Sawyer::Resource] Total commit counts for the owner and total commit
-      #   counts in all. all is everyone combined, including the owner in the last
-      #   52 weeks. If you’d like to get the commit counts for non-owners, you can
-      #   subtract all from owner.
-      # @see https://developer.github.com/v3/repos/statistics/#get-the-weekly-commit-count-for-the-repository-owner-and-everyone-else
-      # @example Get weekly commit counts for octokit
-      #   @client.participation_stats("octokit/octokit.rb")
-      def participation_stats(repo, options = {})
-        get_stats(repo, "participation", options)
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @return [Array<Sawyer::Resource>] A list of stats
+      # @see https://developer.github.com/v3/repos/statistics/#get-contributors-list-with-additions-deletions-and-commit-counts
+      def contributors_stats(repo, options = {})
+        get "#{Repository.path repo}/stats/contributors", options
       end
 
       # Get the number of commits per hour in each day
       #
-      # @param repo [Integer, String, Hash, Repository] A GitHub repository
-      # @option retry_timeout [Number] How long Octokit should keep trying to get stats (in seconds)
-      # @option retry_wait [Number] How long Octokit should wait between retries.
-      # @return [Array<Array>] Arrays containing the day number, hour number, and
-      #   number of commits
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @return [Array<Sawyer::Resource>] A list of stats
       # @see https://developer.github.com/v3/repos/statistics/#get-the-number-of-commits-per-hour-in-each-day
-      # @example Get octokit punch card
-      #   @octokit.punch_card_stats
       def punch_card_stats(repo, options = {})
-        get_stats(repo, "punch_card", options)
-      end
-      alias :punch_card :punch_card_stats
-
-      private
-
-      # @private Get stats for a repository
-      #
-      # @param repo [Integer, String, Hash, Repository] A GitHub repository
-      # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
-      # @see https://developer.github.com/v3/repos/statistics/
-      def get_stats(repo, metric, options = {})
```

Comment:
For out future reference, we're going to recommend using https://github.com/octokit/octopoller.rb instead of our built in functionality.